dev_tools/scripts/format_header.sh

```diff
@@ -120,6 +120,41 @@ get_include_regex () {
     fi
 }
 
+# Test if required commands are present on the system:
+command -v "$CLANG_FORMAT" &> /dev/null
+if [ ${?} -ne 0 ]; then
+    echo "The command 'clang-format' is required for this script to work, but not supported by your system. It can be set via environment parameter CLANG_FORMAT=<clang-format path>" 1>&2
+    exit 1
+fi
+
+# Test the command on MacOS
+declare -n &> /dev/null
+if [ ${?} -ne 0 ]; then
+    echo "The command 'declare' is required to support '-n' option, please update bash or use 'brew install bash' if on MacOS" 1>&2
```

Comment:
Nit: Some reframing: 
```suggestion
    echo "The command 'declare' needs to support the '-n' option, Please update bash or use 'brew install bash' if on MacOS" 1>&2
```